client/messaging/message-input.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useCallback, useRef, useState } from 'react'
+import styled from 'styled-components'
+import KeyListener from '../keyboard/key-listener'
+import TextField from '../material/text-field'
+import { colorDividers } from '../styles/colors'
+
+const StyledTextField = styled(TextField)<{ showDivider?: boolean }>`
+  flex-shrink: 0;
+  position: relative;
+  padding: 8px 16px;
+  contain: content;
+
+  &::after {
+    position: absolute;
+    height: 1px;
+    left: 0px;
+    right: 0px;
+    top: 0;
+
+    content: '';
+    border-top: 1px solid ${props => (props.showDivider ? colorDividers : 'transparent')};
+    transition: border 250ms linear;
+  }
+`
+
+export interface MessageInputProps {
+  className?: string
+  showDivider?: boolean
+  onSendChatMessage: (msg: string) => void
+}
+
+export default function MessageInput(props: MessageInputProps) {
+  const [message, setMessage] = useState<string>('')
+  const inputRef = useRef<TextField | null>(null)
```

Comment:
I *think* the `useRef` will already include `null` in the possible types if you pass null as the parameter, so you could just do:

```ts
const inputRef = useRef<TextField>(null)
```

(in VSCode, at least, you can mouse over these vars and see what type has been assigned to them)